src/editor/ui_menus/tool_change_resources_options_menu.cc

```diff
@@ -88,7 +88,7 @@ EditorToolChangeResourcesOptionsMenu::EditorToolChangeResourcesOptionsMenu(
 	// Add resource buttons
 	resources_box_.add_inf_space();
 	const Widelands::Descriptions& descriptions = parent.egbase().descriptions();
-	for (Widelands::DescriptionIndex i = 0; i < descriptions.nr_resources(); ++i) {
+	for (size_t i = 0; i < descriptions.nr_resources(); ++i) {
```

Comment:
I agree with you that using the WL data type would be preferable, but I didn't know of a better or more effective way of fixing the comparison error that codeql complained about.

The only other way I know to fix that would be to change how Widelands::DescriptionIndex was declared. I'm not familiar enough with the codebase to know whether or not that would be worth it. I assume there'd be a "cascade" effect where many other changes would then be needed related to how it's typed.

Reply:
casting the comparison value to a narrower type might do the trick in this case as it would prevent the danger of an infinte loop which is the cause for this warning.

Reply:
See https://github.com/widelands/widelands/pull/5342#discussion_r845573861